src/states_screens/race_gui_overworld.cpp

```diff
@@ -321,18 +331,62 @@ void RaceGUIOverworld::drawTrophyPoints()
         font->draw(hardTrophiesW.c_str(), dest, time_color, false, vcenter, NULL, true /* ignore RTL */);
     }
 
-    dest = core::rect<s32>(pos.UpperLeftCorner.X - size - 5, pos.UpperLeftCorner.Y,
-                           pos.UpperLeftCorner.X - 5, pos.UpperLeftCorner.Y + size);
+    dest += core::position2di(size*2, 0);
+    if (!m_close_to_a_challenge && !PlayerManager::getCurrentPlayer()->isLocked("difficulty_best"))
+    {
+        draw2DImage(m_trophy4, dest, source, NULL, NULL, true /* alpha */);
+    }
+    dest += core::position2di((int)(size*1.5f), 0);
+    std::string bestTrophies = StringUtils::toString(player->getNumBestTrophies());
+    core::stringw bestTrophiesW(bestTrophies.c_str());
+    if (!m_close_to_a_challenge && !PlayerManager::getCurrentPlayer()->isLocked("difficulty_best"))
+    {
+        font->draw(bestTrophiesW.c_str(), dest, time_color, false, vcenter, NULL, true /* ignore RTL */);
+    }
+
+    dest = core::rect<s32>(pos.UpperLeftCorner.X - size, pos.UpperLeftCorner.Y,
+                           pos.UpperLeftCorner.X, pos.UpperLeftCorner.Y + size);
 
     draw2DImage(m_open_challenge, dest, source, NULL,
                                               NULL, true /* alpha */);
 
-    pos.LowerRightCorner.Y = dest.LowerRightCorner.Y;
-    pos.UpperLeftCorner.X += 5;
+    core::dimension2du area = font->getDimension(L"9");
+    int small_width = area.Width;
+    area = font->getDimension(L"99");
+    int middle_width = area.Width;
+    area = font->getDimension(L"999");
+    int large_width = area.Width;
+
+    int number_width;
 
+    if (points < 9) number_width = small_width;
+    else if (points <99) number_width = middle_width;
+    else number_width = large_width;
+
+    pos.LowerRightCorner.Y = dest.LowerRightCorner.Y + 1.5f*size;
+    pos.UpperLeftCorner.X -= (0.5f*size + number_width*0.5f);
 
     font->draw(sw.c_str(), pos, time_color, false, vcenter, NULL, true /* ignore RTL */);
-    font->disableShadow();
+
+    pos.UpperLeftCorner.X += (0.5f*size + number_width*0.5f);
+
+    if (next_unlock_points > points && (points + 80) >= next_unlock_points)
+    {
+        if (next_unlock_points < 9) number_width = small_width;
+        else if (next_unlock_points <99) number_width = middle_width;
+        else number_width = large_width;
+
+        dest = core::rect<s32>(pos.UpperLeftCorner.X - 2.5f*size, pos.UpperLeftCorner.Y,
+                           pos.UpperLeftCorner.X - 1.5f*size, pos.UpperLeftCorner.Y + size);
+
+        draw2DImage(m_locked_bonus, dest, source, NULL,
+                                                  NULL, true /* alpha */);
+
+        pos.UpperLeftCorner.X -= (2*size + number_width*0.5f);
+
+        font->draw(swg.c_str(), pos, time_color, false, vcenter, NULL, true /* ignore RTL */);
+        font->disableShadow();
```

Comment:
font->disableShadow has been moved inside a "if", whereas before it was always executed. I expect that's why

Reply:
That would also explain why I didn't see the issue when trying to reproduce it, as this if returned "true" in my test situation.